components/button/css/template.tsx

```diff
@@ -84,66 +89,70 @@ export const Button: React.FC<IButtonProps> = ({
   focusVisible = defaultArgs.focusVisible,
   hover = defaultArgs.hover,
   textContent = defaultArgs.textContent,
+  children,
   showIcon = defaultArgs.showIcon,
   icon = defaultArgs.icon,
   classNames = [],
 }) => {
-  const iconMarkup = (
-    <div className={clsx(`rvo-button__icon--${kind}-action`, 'rvo-icon', `rvo-icon-${icon}`, 'rvo-icon--md')}></div>
-  );
+  const iconMarkup = <Icon icon={icon} size="md" />;
   return (
-    <button
+    <ButtonUtrecht
       className={clsx(
-        'utrecht-button',
         classNames,
-        kind === 'primary' && 'utrecht-button--primary-action',
-        kind === 'secondary' && 'utrecht-button--secondary-action',
-        kind === 'tertiary' && 'rvo-button--tertiary-action',
-        kind === 'quaternary' && 'rvo-button--quaternary-action',
-        kind === 'warning-subtle' && 'rvo-button--warning-subtle-action',
-        kind === 'warning' && 'rvo-button--warning-action',
+        kind === 'tertiary' && 'utrecht-button--rvo-tertiary-action',
+        kind === 'quaternary' && 'utrecht-button--rvo-quaternary-action',
         active && 'utrecht-button--active',
         busy && 'utrecht-button--busy',
         hover && 'utrecht-button--hover',
         focus && 'utrecht-button--focus',
         focusVisible && 'utrecht-button--focus-visible',
-        disabled && 'utrecht-button--disabled',
         'rvo-layout-row',
         size === 'xs' && 'rvo-layout-gap--xs',
         size === 'sm' && 'rvo-layout-gap--sm',
         size === 'md' && 'rvo-layout-gap--md',
-        size === 'xs' && 'rvo-button--xs',
-        size === 'sm' && 'rvo-button--sm',
-        size === 'md' && 'rvo-button--md',
+        size === 'xs' && 'utrecht-button--rvo-xs',
+        size === 'sm' && 'utrecht-button--rvo-sm',
+        size === 'md' && 'utrecht-button--rvo-md',
       )}
-      aria-disabled={disabled || null}
+      disabled={disabled || null}
+      appearance={
+        kind === 'primary' || kind === 'warning'
+          ? 'primary-action-button'
+          : kind === 'secondary'
+          ? 'secondary-action-button'
+          : kind === 'subtle' || kind === 'warning-subtle'
+          ? 'subtle-button'
+          : null
+      }
+      hint={kind === 'warning' || kind === 'warning-subtle' ? 'warning' : null}
     >
       {showIcon === 'before' && iconMarkup}
```

Comment:
I think it might have to do with `children?: React.ReactNode;`, allowing one instead of many children. I have updated all the property definitions to use `PropsWithChildren` from React to fix this.